pkg/storagev2/inmemstore/inmemstore.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2022 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package inmem
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/armon/go-radix"
```

Comment:
is this the hashicorp one?

Reply:
Yes. Hashicorp has a immutable-radix implementation. In that they have linked this as the mutable version. This is what we need. Its a very good and stable library I think.